### PR TITLE
#170044697 Signup Endpoint

### PR DESCRIPTION
**Pull request by Elvis-rugamba**
#### What does this PR do?
Have Signup Endpoint
#### Description of Task to be completed?
Have the Signup Endpoint working: `api/v2/auth/signup`
#### How should this be manually tested?
- `npm test`
- Using postman to test endpoint
#### Any background context you want to provide?
There are babel plugins installed to support async/await
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/170044697

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -28,9 +27,8 @@ app.use((req, res, next) => {
 });
 
 app.use('/api/v2/auth', userRoutes);
-app.use('/api/v2/red-flags', redFlagRoutes);
-app.use('/api/v2/api-docs', swaggerUi.serve,
-  swaggerUi.setup(swaggerDocument, swaggerOptions));
+/* app.use('/api/v2/api-docs', swaggerUi.serve,
+  swaggerUi.setup(swaggerDocument, swaggerOptions)); */
```
Comment (dusmel) [inline server/app.js]: Kindly remove commented code.

File: server/v2/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+      const user = {
+        firstname: req.body.firstname,
+        lastname: req.body.lastname,
+        email: req.body.email,
+        phonenumber: req.body.phoneNumber,
+        username: req.body.username,
+        password: HashedPassword,
+        type: req.body.type,
+      };
```
Comment (dusmel) [inline server/v2/controllers/user.js]: You can use destructuring here.

File: server/v2/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+      const createdUser = await User.create(user);
+
+      const token = GenerateToken.getToken(createdUser);
+
+      res.status(201).json({
+        status: 201,
+        message: 'User created successfully',
+        data: { token: token },
```
Comment (dusmel) [inline server/v2/controllers/user.js]: Try to follow the format provided on ADC.

File: server/v2/test/user.spec.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        expect(res).to.have.status(404);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('error');
+        expect(res.body.status).to.be.eql(404);
+        done();
+      })
+      .catch((err) => done(err));
+  });
+});
```
Comment (dusmel) [inline server/v2/test/user.spec.js:109]: What is this block testing?

**Review by dusmel — CHANGES_REQUESTED** [review]
Goog job @Elvis-rugamba, super clean code, keep it up! Kindly look at the comments.

File: server/v2/test/user.spec.js (hunk 4 above)
Comment (Elvis-rugamba) [reply server/v2/test/user.spec.js:109]: Is testing if you didn't entered a valid url

Comment (Elvis-rugamba) [reply server/v2/test/user.spec.js:109]: And it returns 404

**Review by dusmel — APPROVED** [review]

**Pull request merged**